src/Build/Definition/Project.cs

```diff
@@ -2782,17 +2782,18 @@ private List<ProvenanceResult> GetItemProvenance(string itemToMatch, IEnumerable
                     return new List<ProvenanceResult>();
                 }
 
-                return
-                    projectItemElements
+                return projectItemElements
                     .AsParallel()
-                    .AsOrdered()
-                    .Select(i => ComputeProvenanceResult(itemToMatch, i))
+                    .Select((item, index) => ComputeProvenanceResult(itemToMatch, item, index))
```

Comment:
Wondering if creating the tuple here wouldn't be tiny bit more readable. Just a nit really, feel free to ignore.

```suggestion
                    .Select((item, index) => (Result: ComputeProvenanceResult(itemToMatch, item), Index: index))
```

Reply:
Good advice. This allows to revert complexity I added to `ComputeProvenanceResult` :)